resources/datastore.go

```diff
@@ -0,0 +1,441 @@
+				Description: "Enable or disable notifications. Default is false",
+			},
+			"notifications_emails": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "List of email addresses to send notifications to",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			// "maintenance_settings": {
+			// 	Type:        schema.TypeMap,
+			// 	Optional:    true,
+			// 	Description: "Maintenance settings",
+			// 	Elem:        (maintenance{}).Schema(),
+			// },
+			"maintenance_day_of_week": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: "Day of the week to run the maintenance. 1-7, 1 is Monday",
+			},
+			"maintenance_start_hour": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: "Hour of the day to start the maintenance. 0-23",
+			},
+			"maintenance_end_hour": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: "Hour of the day to end the maintenance. 0-23. Must be start_hour + 2",
+			},
+		},
+		Create: r.Create,
+		Read:   r.Read,
+		Update: r.Update,
+		Delete: r.Delete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+	}
+}
+
+func (r *Datastore) Create(d *schema.ResourceData, _ any) error {
+	ctx := context.Background()
+	c, err := schemaToDatastore(d)
+
+	if err != nil {
+		return err
+	}
+
+	n, err := r.svc.Create(ctx, c)
+	if err != nil {
+		d.SetId("")
+		return fmt.Errorf("creating stores: %w", err)
+	}
+
+	var errs []error
+
+	if len(c.DbParams) != 0 {
+		if err := r.svc.SetParameters(ctx, n.ID, c.DbParams); err != nil {
+			errs = append(errs, fmt.Errorf("%w setting: %w", ccx.ParametersErr, err))
+		} else {
+			n.DbParams = c.DbParams
+		}
+	}
+
+	if err := r.svc.SetFirewallRules(ctx, n.ID, c.FirewallRules); err != nil {
+		errs = append(errs, fmt.Errorf("%w: setting: %w", ccx.FirewallRulesErr, err))
+	} else {
+		n.FirewallRules = c.FirewallRules
+	}
+
+	if err := schemaFromDatastore(*n, d); err != nil {
+		errs = append(errs, fmt.Errorf("setting schema: %w", err))
+	}
+
+	if len(errs) != 0 {
+		return fmt.Errorf("creating stores completed only partially: %w", errors.Join(errs...))
+	}
+
+	return nil
+}
+
+func (r *Datastore) Read(d *schema.ResourceData, _ any) error {
+	ctx := context.Background()
+	c, err := schemaToDatastore(d)
+
+	if err != nil {
+		return err
+	}
+
+	n, err := r.svc.Read(ctx, c.ID)
+	if errors.Is(err, ccx.ResourceNotFoundErr) {
+		d.SetId("")
+		return nil
+	} else if err != nil {
+		return err
+	}
+
+	return schemaFromDatastore(*n, d)
+}
+
+func (r *Datastore) Update(d *schema.ResourceData, _ any) error {
+	ctx := context.Background()
+
+	c, err := schemaToDatastore(d)
+	if err != nil {
+		return err
+	}
+
+	old, err := r.svc.Read(ctx, c.ID)
+	if err != nil {
+		return err
+	}
+
+	n, err := r.svc.Update(ctx, *old, c)
+	if err != nil {
+		return err
+	}
+
+	var errs []error
+
+	if err := r.svc.SetParameters(ctx, n.ID, c.DbParams); err != nil {
```

Comment:
this happens unconditionally, probably worth checking if the parameters have changed?